File: app/schemas/prescription.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class PrescriptionCreate(BaseModel):
    medical_record_id: int
    medication: str
    dosage: str
    frequency: str
    duration: str
    instructions: str
    
class PrescriptionResponse(BaseModel):
    id: int
    patient_id: int
    doctor_id: int
    medical_record_id: int
    medication: str
    dosage: str
    frequency: str
    duration: str
    instructions: str
    created_at: datetime
    
class PrescriptionUpdate(BaseModel):
    medication: Optional[str] = None
    dosage: Optional[str] = None
    frequency: Optional[str] = None
    duration: Optional[str] = None
    instructions: Optional[str] = None

File: frontend/src/types/prescription.ts
export interface Prescription {
  id: number;
  patient_id: number;
  doctor_id: number;
  medical_record_id: number;
  medication: string;
  dosage: string;
  frequency: string;
  duration: string;
  instructions: string;
  created_at: string;
}

export interface PrescriptionCreate {
  medical_record_id: number;
  medication: string;
  dosage: string;
  frequency: string;
  duration: string;
  instructions: string;
}

export interface PrescriptionUpdate {
  medication?: string;
  dosage?: string;
  frequency?: string;
  duration?: string;
  instructions?: string;
}